src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEM.cpp

```diff
@@ -0,0 +1,776 @@
+                                       extrinsicMeshData::Displacementy_np1,
+                                       extrinsicMeshData::Displacementz_np1,
+                                       extrinsicMeshData::ForcingRHSx,
+                                       extrinsicMeshData::ForcingRHSy,
+                                       extrinsicMeshData::ForcingRHSz,
+                                       extrinsicMeshData::MassVector,
+                                       extrinsicMeshData::DampingVectorx,
+                                       extrinsicMeshData::DampingVectory,
+                                       extrinsicMeshData::DampingVectorz,
+                                       extrinsicMeshData::StiffnessVectorx,
+                                       extrinsicMeshData::StiffnessVectory,
+                                       extrinsicMeshData::StiffnessVectorz,
+                                       extrinsicMeshData::FreeSurfaceNodeIndicator >( this->getName() );
+
+    FaceManager & faceManager = mesh.getFaceManager();
+    faceManager.registerExtrinsicData< extrinsicMeshData::FreeSurfaceFaceIndicator >( this->getName() );
+
+    ElementRegionManager & elemManager = mesh.getElemManager();
+
+    elemManager.forElementSubRegions< CellElementSubRegion >( [&]( CellElementSubRegion & subRegion )
+    {
+      subRegion.registerExtrinsicData< extrinsicMeshData::MediumVelocityVp >( this->getName() );
+      subRegion.registerExtrinsicData< extrinsicMeshData::MediumVelocityVs >( this->getName() );
+      subRegion.registerExtrinsicData< extrinsicMeshData::MediumDensity >( this->getName() );
+    } );
+
+  } );
+}
+
+
+
+void ElasticWaveEquationSEM::postProcessInput()
+{
+
+  GEOSX_ERROR_IF( m_sourceCoordinates.size( 1 ) != 3,
+                  "Invalid number of physical coordinates for the sources" );
+
+  GEOSX_ERROR_IF( m_receiverCoordinates.size( 1 ) != 3,
+                  "Invalid number of physical coordinates for the receivers" );
+
+  EventManager const & event = this->getGroupByPath< EventManager >( "/Problem/Events" );
+  real64 const & maxTime = event.getReference< real64 >( EventManager::viewKeyStruct::maxTimeString() );
+  real64 dt = 0;
+  for( localIndex numSubEvent = 0; numSubEvent < event.numSubGroups(); ++numSubEvent )
+  {
+    EventBase const * subEvent = static_cast< EventBase const * >( event.getSubGroups()[numSubEvent] );
+    if( subEvent->getEventName() == "/Solvers/" + this->getName() )
+    {
+      dt = subEvent->getReference< real64 >( EventBase::viewKeyStruct::forceDtString() );
+    }
+  }
+
+  GEOSX_THROW_IF( dt < epsilonLoc*maxTime, "Value for dt: " << dt <<" is smaller than local threshold: " << epsilonLoc, std::runtime_error );
+
+  if( m_dtSeismoTrace > 0 )
+  {
+    m_nsamplesSeismoTrace = int( maxTime / m_dtSeismoTrace) + 1;
+  }
+  else
+  {
+    m_nsamplesSeismoTrace = 0;
+  }
+  localIndex const nsamples = int(maxTime/dt) + 1;
```

Comment:
This can be moved to the `postProcessInput` function of the base class, and then you will be able to call `WaveSolverBase::postProcessInput` both here and in `AcousticWaveEquationSEM`, which will remove a lot of code duplication.